app/services/locations/finder.rb

```diff
@@ -24,7 +24,8 @@ def apply_filters(scope)
     def apply_supplier_filters(scope)
       return scope unless filter_params.key?(:supplier_id)
 
-      scope.joins(:locations_suppliers).where(suppliers: { id: filter_params[:supplier_id] })
+      scope = scope.where(suppliers: { id: filter_params[:supplier_id] })
+      scope.merge(SupplierLocation.effective_on(Time.zone.today))
```

Comment:
Not sure if this should be here 🤔 
maybe just rename the method to `apply_effective_supplier_filters`

Reply:
The locations for a given supplier depend on the effective date - as this is a reference data endpoint it should return the current set of locations for the supplier. Then when suppliers call this before/after the go live date they receive the correct set of locations according to their contract.